transaction/core/src/ring_ct/rct_bulletproofs.rs

```diff
@@ -148,15 +162,22 @@ impl SigningData {
     /// * `check_value_is_preserved` - If true, check that the value of inputs
     ///   equals value of outputs.
     /// * `rng` - randomness
+    ///
+    /// Returns:
+    /// * The SigningData
+    /// * The TxSummary
+    /// * The extended message digest
+    ///
+    /// (The latter two are MCIP #52 and only relevant on small HW wallets)
     pub fn new<CSPRNG: RngCore + CryptoRng>(
```

Comment:
Optional/Equivalent: Rather than changing the return type of `new` and all its callers, add a `new_with_summary` that returns all 3, and update `new` to `new_with_summary(...).0`

Reply:
I think it is consistent with a lot of our other implementations, e.g. `ConsensusServiceEnclave`. but it's a pretty minor change, i'm okay to do this if you want.

to some extent I want to smoke out the implementations and make sure they do something with these extra things, at least in debug mode. so that's a good reason to keep it this way for the moment.

Reply:
Ah OK, I thought the other impls were intentionally not using those bits. It sounds like you're planning on utilizing the returned values elsewhere, so feel free to ignore the suggestion

Reply:
I think you guys are right and new_with_summary is better (having now actually developed the next PR).
Gonna follow that route now for this change